### PR TITLE
Retrieve

### PR DESCRIPTION
**Pull request by andrewMacmurray**
+ adds class for interacting with fileSystem
+ adds commands registry that coordinates communication with command socket and file socket
+ user can issue a `RETR` command and receive the correct responses and file

### CODE REVIEW

File: src/main/java/filesystem/NativeFileSystem.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    public boolean fileExists(String path) {
+        return Files.exists(resolveRoot(path));
+    }
+
+    public InputStream readFile(String path) throws IOException {
+        return Files.newInputStream(resolveRoot(path));
+    }
+
+    public void writeFile(String destinationPath, InputStream source) throws IOException {
+        Files.copy(source, resolveRoot(destinationPath));
```
Comment (fabientownsend) [inline src/main/java/filesystem/NativeFileSystem.java]: Most of these system manipulations can raise an exception, you currently throw up these exception up to the _controller_ is that on purpose? It mean that currently if something go wrong, the only clue that you will have about what happened it the code + error message that you picked up a this very top level

File: src/main/java/server/CommandRegistry.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        );
+    }
+
+    private void STOR(String fileName) {
+        try {
+            statusResponder.respond(150, "OK receiving file");
+            fileHandler.store(fileName);
+            statusResponder.respond(250, "OK File stored");
+        } catch (IOException e) {
+            statusResponder.respond(450, "Something went wrong");
```
Comment (fabientownsend) [inline src/main/java/server/CommandRegistry.java]: `450 Blocked by Windows Parental Controls (Microsoft)`
First time I see it

Comment (andrewMacmurray) [reply src/main/java/server/CommandRegistry.java]: @fabientownsend lol I got `450` from here http://www.serv-u.com/kb/1522/450-FTP-Reply-Code, but I've yet to go through the formalised response codes yet (just getting it working for now)

File: src/main/java/filesystem/NativeFileSystem.java (hunk 1 above)
Comment (andrewMacmurray) [reply src/main/java/filesystem/NativeFileSystem.java]: @fabientownsend I haven't quite thought that far yet but I was thinking of catching them in the relevant `CommandRegistry` handles so I can indicate to the client that something went wrong with them. What were you thinking as an alternative?

File: src/main/java/server/CommandRegistry.java (hunk 2 above)
Comment (fabientownsend) [reply src/main/java/server/CommandRegistry.java]: OH right, FTP protocol, nevermind! :D

**Pull request merged**